### PR TITLE
[BEAM-1593] Remove Pipeline.getRunner

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Runners need not be instantiated until after pipeline construction, so
they should not be exposed by the Pipeline class.

### CODE REVIEW

**Comment by davorbonaci** [comment]
R: @kennknowles

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java [hunk 1]
```diff
@@ -361,6 +314,52 @@ public DirectPipelineResult run(Pipeline pipeline) {
...
+   * The default set of transform overrides to use in the {@link DirectRunner}.
+   *
+   * <p>The order in which overrides is applied is important, as some overrides are expanded into a
+   * composite. If the composite contains {@link PTransform PTransforms} which are also overridden,
+   * these PTransforms must occur later in the iteration order. {@link ImmutableMap} has an
+   * iteration order based on the order at which elements are added to it.
+   */
+  @SuppressWarnings("rawtypes")
+  private Map<PTransformMatcher, PTransformOverrideFactory> defaultTransformOverrides() {
+    return ImmutableMap.<PTransformMatcher, PTransformOverrideFactory>builder()
```
Comment (kennknowles) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java]: Noting that I do agree with Eugene that this should be a data structure with more explicit order than `Map`.

**Comment by kennknowles** [comment]
Run Dataflow RunnableOnService

**Comment by kennknowles** [comment]
The failure is a semantic conflict that did not manifest as a code conflict - I added two tests that follow the pattern you modified. Please update those and start another ROS run.

**Comment by tgroh** [comment]
Updated

**Comment by kennknowles** [comment]
Run Dataflow RunnableOnService

**Comment by kennknowles** [comment]
OK, tests happy and LGTM.

**Pull request merged**